scylla-server-rust/src/processors/mqtt_processor.rs

```diff
@@ -0,0 +1,204 @@
+                            self.send_db_msg(msg.clone()).await;
+                            self.send_socket_msg(msg).await;
+                        }
+
+            },
+                    _ = spec_interval.tick() => {
+                        trace!("Updating viewership data!");
+                        if let Ok(sockets) = self.io.sockets() {
+                        let client_data = ClientData {
+                            name: "Viewers".to_string(),
+                            node: "Internal".to_string(),
+                            unit: "".to_string(),
+                            run_id: self.curr_run,
+                            timestamp: chrono::offset::Utc::now().timestamp_millis(),
+                            values: vec![sockets.len().to_string()]
+                        };
+                        self.send_socket_msg(client_data).await;
+
+                    } else {
+                        warn!("Could not fetch socket count");
+                    }
+                    }
+
+                }
+        }
+    }
+
+    /// Parse the message
+    /// * `msg` - The mqtt message to parse
+    /// returns the ClientData, or the Err of something that can be debug printed
+    #[instrument(skip(self), level = Level::TRACE)]
+    fn parse_msg(&self, msg: Publish) -> Result<ClientData, impl fmt::Debug> {
+        let data = serverdata::ServerData::parse_from_bytes(&msg.payload)
+            .map_err(|f| format!("Could not parse message topic:{:?} error: {}", msg.topic, f))?;
+
+        let split = std::str::from_utf8(&msg.topic)
+            .unwrap_or_else(|_| panic!("Could not parse topic: {:?}", msg.topic))
+            .split_once('/')
+            .ok_or(&format!("Could not parse nesting: {:?}", msg.topic))?;
+
+        let node = split.0;
+
+        let data_type = split.1.replace('/', "-");
+
+        // extract the unix time, returning the current time instead if needed
+        // note the Cow magic involves the return from the map is a borrow, but the unwrap cannot as we dont own it
+        let unix_time = msg
+            .properties
+            .unwrap_or_default()
+            .user_properties
+            .iter()
+            .map(Cow::Borrowed)
+            .find(|f| f.0 == "ts")
+            .unwrap_or_else(|| {
+                debug!("Could not find timestamp in mqtt, using current time");
+                Cow::Owned((
+                    "ts".to_string(),
+                    chrono::offset::Utc::now().timestamp_millis().to_string(),
+                ))
+            })
+            .into_owned();
+
+        // parse time, if time isnt present use sys time (see above)
+        let Ok(time_clean) = unix_time.1.parse::<i64>() else {
+            return Err(format!("Invalid timestamp: {}", unix_time.1));
+        };
+        // ts check for bad sources of time which may return 1970
+        if time_clean < 963014966000 {
+            return Err(format!("Timestamp before year 2000: {}", unix_time.1));
```

Comment:
same here

Reply:
What if router time is 1970 also?  That's what this logic is for.

Reply:
Then L ig i mean add another check for it